src/main/java/com/rcv/Tabulator.java

```diff
@@ -234,58 +234,70 @@ private void logSummaryInfo() {
   // logic only considers continuing candidates, so it won't assign any votes to past winners -- but
   // in reality they continue to hold their winning margins for the rest of the rounds, so we need
   // to fill in those values here.
-  // TODO: instead of recomputing these winning tallies each round, we could just read the values
-  // from the previous round. Once someone wins and we redistribute their extra votes, their tally
-  // should be frozen for the rest of the rounds.
+  // We need to do the computation once for each winner to account for transfers. In subsequent
+  // rounds, we can just copy the number from the previous round, since it won't change.
   private void updatePastWinnerTallies() {
-    // pastWinners contains winners from rounds that preceded the current round
-    Set<String> pastWinners = new HashSet<>();
+    Map<String, BigDecimal> roundTally = roundTallies.get(currentRound);
+    Map<String, BigDecimal> previousRoundTally = roundTallies.get(currentRound - 1);
```

Comment:
What prevents this from bugging out in the first round, since it presumably does a .get(0)?

Reply:
`.get(0)` is fine in itself -- it just returns null. And in that case, we're not going to end up trying to use the null later in the method, because `winnersToProcess` is guaranteed to be empty.